gnome-image-installer/pages/confirm/gis-confirm-page.h

```diff
@@ -0,0 +1,61 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ */
+
+#ifndef __GIS_CONFIRM_PAGE_H__
+#define __GIS_CONFIRM_PAGE_H__
+
+#include "gnome-image-installer.h"
+
+G_BEGIN_DECLS
+
+#define GIS_TYPE_CONFIRM_PAGE               (gis_confirm_page_get_type ())
+#define GIS_CONFIRM_PAGE(obj)                           (G_TYPE_CHECK_INSTANCE_CAST ((obj), GIS_TYPE_CONFIRM_PAGE, GisConfirmPage))
+#define GIS_CONFIRM_PAGE_CLASS(klass)                   (G_TYPE_CHECK_CLASS_CAST ((klass),  GIS_TYPE_CONFIRM_PAGE, GisConfirmPageClass))
+#define GIS_IS_CONFIRM_PAGE(obj)         (G_TYPE_CHECK_INSTANCE_TYPE ((obj), GIS_TYPE_CONFIRM_PAGE))
+#define GIS_IS_CONFIRM_PAGE_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass),  GIS_TYPE_CONFIRM_PAGE))
+#define GIS_CONFIRM_PAGE_GET_CLASS(obj)                 (G_TYPE_INSTANCE_GET_CLASS ((obj),  GIS_TYPE_CONFIRM_PAGE, GisConfirmPageClass))
+
+typedef struct _GisConfirmPage        GisConfirmPage;
+typedef struct _GisConfirmPageClass   GisConfirmPageClass;
+
+struct _GisConfirmPage
+{
+  GisPage parent;
+};
```

Comment:
Nitpick: Use `G_DECLARE_FINAL_TYPE` and BANG the boilerplate is gone.

(This is just a nitpick, since I assume this has all been cargo-culted, and I can see why you might want to keep it consistent with other pages and port them all at a later date. Also, what you have is not wrong.)

Reply:
Yes, it's copy-pasta. On the other hand, I did make this page use `gtk_widget_class_set_template_from_resource` which no other page uses, so I could have updated this at the same time… oh well.